### ISSUE TITLE
No warning if constant is null or handler is missing

### ISSUE THREAD

**Issue by jhollingworth** [jhollingworth/marty#272]
This code will produce the unhelpful error "TypeError: Cannot read property 'isActionCreator' of null "

```js
var FooStore = Marty.createStore({
  handlers: {
    bar: null
  },
  bar() {
  }
});
```

and this code will cause the page to render nothing

```js
var BarStore = Marty.createStore({
  handlers: {
    bar: "BAR"
  }
});
```

This is a regression as there used to be useful warnings for it

**Comment by taion** [comment]
I took a quick look here. Some interesting things going on.

The `null` issue happens for an interesting reason. The current store implementation only validates the handlers in `handleActions`, so you don't even potentially get the warning until the first time something gets dispatched (rather than on store creation... but you can't really check on store creation because the base class constructor is invoked first, and `handlers` isn't currently static).

Then there's another interesting thing going on with

```js
_.once(() => validateHandlers(this))
```

where as implemented, `_.once` really only does run that function once, even if it throws, and if it does throw, it will just keep returning undefined whenever you call it, which means that you only throw the `ActionPredicateUndefinedError` the first time, but then pass right through `validateHandlers` on subsequent calls, which gets you to the `isActionCreator` error.

The last wrinkle is that, because of the validation only getting invoked on action handling, if you have code that looks like the example here: http://martyjs.org/guides/queries/index.html, that `catch` will actually catch the exception thrown from `this.dispatch`, which then won't necessarily get logged out. Without it, the `ActionPredicateUndefinedError` does get logged (the first time, subsequent times log the error on `isActionCreator`).

**Comment by sporto** [comment]
I had an action with:

```
	this.dispatch(constants.NOTES_UPDATED, data);
```

NOTES_UPDATED was undefined. Marty would happily try to dispatch this. I would probably expect to see an error telling me that NOTES_UPDATED is undefined.

**Comment by jhollingworth** [comment]
@sporto good point, I've just [added a check for that](https://github.com/martyjs/marty-lib/commit/3877ca6f1a4d3cae3754ff1841fa1c1c2a16cee3)
